src/containers/sb-file-uploader.jsx

```diff
@@ -60,35 +87,41 @@ class SBFileUploader extends React.Component {
     // called when user has finished selecting a file to upload
     handleChange (e) {
-        // Remove the hash if any (without triggering a hash change event or a reload)
-        history.replaceState({}, document.title, '.');
-        const reader = new FileReader();
         const thisFileInput = e.target;
-        reader.onload = () => this.props.vm.loadProject(reader.result)
-            .then(() => {
-                analytics.event({
-                    category: 'project',
-                    action: 'Import Project File',
-                    nonInteraction: true
-                });
-                this.props.onLoadingFinished(this.props.loadingState);
-                // Reset the file input after project is loaded
-                // This is necessary in case the user wants to reload a project
-                thisFileInput.value = null;
-            })
-            .catch(error => {
-                log.warn(error);
-                alert(this.props.intl.formatMessage(messages.loadError)); // eslint-disable-line no-alert
-                this.props.onLoadingFinished(this.props.loadingState);
-                // Reset the file input after project is loaded
-                // This is necessary in case the user wants to reload a project
-                thisFileInput.value = null;
-            });
         if (thisFileInput.files) { // Don't attempt to load if no file was selected
+            this.fileToUpload = thisFileInput.files[0];
+            this.props.requestProjectUpload(this.props.loadingState);
+        }
+    }
+    // called when file upload raw data is available in the reader
+    onload () {
+        if (this.reader) {
             this.props.onLoadingStarted();
-            reader.readAsArrayBuffer(thisFileInput.files[0]);
-            const uploadedProjectTitle = this.getProjectTitleFromFilename(thisFileInput.files[0].name);
-            this.props.onUpdateProjectTitle(uploadedProjectTitle);
+            this.props.vm.loadProject(this.reader.result)
+                .then(() => {
+                    analytics.event({
+                        category: 'project',
+                        action: 'Import Project File',
+                        nonInteraction: true
+                    });
+                    // Remove the hash if any (without triggering a hash change event or a reload)
+                    history.replaceState({}, document.title, '.');
+                    this.props.onLoadingFinished(this.props.loadingState, true);
+                    // Reset the file input after project is loaded
+                    // This is necessary in case the user wants to reload a project
+                    this.resetFileInput();
+                    // disable title handling for now; @todo reenable, with correct logic
+                    // const uploadedProjectTitle = this.getProjectTitleFromFilename(thisFileInput.files[0].name);
+                    // this.props.onUpdateProjectTitle(uploadedProjectTitle);
```

Comment:
Why can't this line be uncommented?

Reply:
I was trying to keep this change focused to fixing this one issue and not the title-switching one; I can't keep the current functionality. But let me try fixing that too, here -- it's probably not so hard